test/puppetlabs/trapperkeeper/services/watcher/filesystem_watch_service_test.clj

```diff
@@ -385,3 +385,21 @@
        (is (logged?
             #"shutdown-on-error triggered because of exception"
             :error))))))
+
+(deftest ^:integration overflow-test
+  (let [watch-dir (fs/temp-dir "overflow")
+        results (atom [])
+        callback (make-callback results)]
+    (with-app-with-config
+     app watch-service-and-deps {}
+     (let [service (tk-app/get-service app :FilesystemWatchService)]
+       (watch! service watch-dir callback))
+     ;; From experimenting the buffer seems to fill at 512 events.
+     ;; However the number of events varies by operating system.
+     (let [files (for [n (range 1000)]
+                   (fs/file watch-dir  (str n ".txt")))]
+       (doseq [f files]
+         (spit f "file content"))
+      (let [events #{{:type :unknown
+                      :path nil}}]
+        (is (= events (wait-for-events results events))))))))
```

Comment:
It seems possible we will get a few real events as well as an overflow here?

Is equality or containment the assertion we care about? Could the `contains-events?` function be used here instead of strict equality?